taigit/src/libraries/Taiga.ts

```diff
@@ -78,4 +78,118 @@ task_history(taskId : number) : Promise<Object> {
+ * * {
+ *      state_trans_invalid : Boolean       // true, the task state transition is not valid; false, the task is valid.
+ *      date : number,         // Date and time of history entry in milliseconds since epoch
+ *      user : object,         // Taiga User Object
+ *      status_trans:entry.String[],//Status transition array
+ *
+ * }
+ */
+export async function
+task_assessment_state_trans(taskId : number) : Promise<Object> {
+let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+let output : Array<Object> = [];
+for(let entry of data) {
+    let new_entry = {
+        state_trans_invalid : false,
+        date : new Date(entry.created_at).getTime(),
+        user : entry.user,
+        status_trans:entry.values_diff.status
+    }
+
+    //Only three status transistion is valid
+    //which is  ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    if((new_entry.status_trans[0] != "New"&&new_entry.status_trans[1] != "In progress")
+        &&(new_entry.status_trans[0] != "In progress"&&new_entry.status_trans[1] != "Ready for test")
+        &&(new_entry.status_trans[0] != "Ready for test"&&new_entry.status_trans[1] != "Closed"))
+        new_entry.state_trans_invalid = true;
+    output.push(new_entry);
+}
+
+return output;
+}
+
+
+/**
+ * @summary This call return a task current end status and estimated finished percentage based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task is abnormal,task status transition, date based on task Id
+ * * {
+ *      finished : Boolean           // true, the task is finished.
+ *      fin_per : number            // finished percentage % default [0%,10%,60%,100%]
+ *      end_status : string,       // current task status  ["New", "In progress","Ready for test", "Closed"]
+ *      num_stat : number           // current task status [0 1 2 3] corresponding to string in end_status
+ *
+ * }
+ */
+export async function
+task_assessment_endstate(taskId : number) : Promise<Object> {
+
+    let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741   or 2555550
+    let output : Array<Object> = [];
+    let endstatus : string = "New";
+    for(let entry of data) {
+        let new_entry = {
+            state_trans_invalid : false,
+            date : new Date(entry.created_at).getTime(),
+            user : entry.user,
+            status_trans : entry.values_diff.status
+        }
+        endstatus = new_entry.status_trans[1];
+        output.push(new_entry);
+    }
+
+    let t_finished : boolean;
+    let t_fin_per : number;
+    let t_num_stat : number;
+     if(endstatus != "Closed") {
+         t_finished = false;
+     }
+    else{
+        t_finished = true;
+    }
```

Comment:
`let t_finished : boolean = (endstatus == "Closed")`?

Reply:
OK. If it is the same logic, it is fine.

Reply:
replace this line with your comment, it seems a error/ or I did not get it right. I will go with the conventional way to prevent error.